internal/readme/source.md

```diff
@@ -0,0 +1,170 @@
+and put the tag map into the returned one.
+If there is already a tag map in the current context, it will be replaced.
+
+[embedmd]:# (tags.go newContext)
+
+In order to update an existing tag map, get the tag map from the current context,
+use NewMap and put the new tag map back to the context.
+
+[embedmd]:# (tags.go replaceTagMap)
+
+
+## Stats
+
+### Creating, retrieving and deleting a measure
+
+Create and load measures with units:
+
+[embedmd]:# (stats.go measure)
+
+Retrieve measure by name:
+
+[embedmd]:# (stats.go findMeasure)
+
+Delete measure (this can be useful when replacing a measure by
+another measure with the same name):
+
+[embedmd]:# (stats.go deleteMeasure)
+
+### Creating an aggregation
+
+Currently only 2 types of aggregations are supported. The CountAggregation is used to count
+the number of times a sample was recorded. The DistributionAggregation is used to
+provide a histogram of the values of the samples.
+
+[embedmd]:# (stats.go aggs)
+
+### Create an aggregation window
+
+Currently only 3 types of aggregation windows are supported. The CumulativeWindow
+is used to continuously aggregate the data received.
+The SlidingTimeWindow to aggregate the data received over the last specified time interval.
+The SlidingCountWindow to aggregate the data received over the last specified sample count.
+Currently all aggregation types are compatible with all aggregation windows.
+Later we might provide aggregation types that are incompatible with some windows.
+
+[embedmd]:# (stats.go windows)
+
+### Creating, registering and unregistering a view
+
+Create and register a view:
+
+[embedmd]:# (stats.go view)
+
+Find view by name:
+
+[embedmd]:# (stats.go findView)
+
+Unregister view:
+
+[embedmd]:# (stats.go unregisterView)
+
+Configure the default interval between reports of collected data.
+This is a system wide interval and impacts all views. The default
+interval duration is 10 seconds. Trying to set an interval with
+a duration less than a certain minimum (maybe 1s) should have no effect.
+
+[embedmd]:# (stats.go reportingPeriod)
+
+### Recording measurements
+
+Recording usage can only be performed against already registered measure
+and their registered views. Measurements are implicitly tagged with the
+tags in the context:
+
+[embedmd]:# (stats.go record)
+
+### Retrieving collected data for a view
+
+Users need to subscribe to a view in order to retrieve collected data.
+
+[embedmd]:# (stats.go subscribe)
+
+Subscribed views' data will be exported via the registered exporters.
+
+[embedmd]:# (stats.go registerExporter)
+
+An example logger exporter is below:
+
+[embedmd]:# (stats.go exporter)
+
+### Force collecting data on demand
+
+Even if a view is registered, if it has no subscriber no data
+for it is collected. In order to retrieve data on-demand for
+view, either the view needs to have at least one subscriber
+or the library needs to be instructed explicitly to collect
+data for the desired view.
+
```

Comment:
I'm not sure this description of subscription is still good after #74

Reply:
Done.